src/metabase/lib/field.cljc

```diff
@@ -228,13 +259,18 @@
                               (or join-alias (lib.join.util/current-join-alias field-metadata))))
         display-name       (if join-display-name
                              (str join-display-name " → " field-display-name)
-                             field-display-name)]
+                             field-display-name)
+        temporal-format    (fn [display-name]
+                             (lib.util/format "%s: %s" display-name (-> (name temporal-unit)
+                                                                        (str/replace \- \space)
+                                                                        u/capitalize-en)))
+        bin-format         (fn [display-name]
+                             (lib.util/format "%s: %s" display-name (lib.binning/binning-display-name binning field-metadata)))]
+    ;; temporal unit and binning formatting are only applied if they haven't been applied yet
     (cond
-      temporal-unit (lib.util/format "%s: %s" display-name (-> (name temporal-unit)
-                                                               (str/replace \- \space)
-                                                               u/capitalize-en))
-      binning       (lib.util/format "%s: %s" display-name (lib.binning/binning-display-name binning field-metadata))
-      :else         display-name)))
+      (and temporal-unit (not= display-name (temporal-format humanized-name))) (temporal-format display-name)
+      (and binning       (not= display-name (bin-format humanized-name)))      (bin-format display-name)
```

Comment:
This is definitely fragile. Any better way to track if the temporal/binning formatting has already been applied?